pkg/endpoints/endpoints.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package endpoints
+
+import (
+	"errors"
+	"net/url"
+	"strings"
+)
+
+// ServiceEndpointFormat is used to identify the custom endpoint of IBM Cloud service.
+var ServiceEndpointFormat string
+
+const (
+	// VPCServiceID used to identify VPC service.
+	VPCServiceID = "vpc"
+	// PVSServiceID used to identify PowerVS service.
+	PVSServiceID = "powervs"
+	// RCServiceID used to identify Resource-Controller service.
+	RCServiceID = "rc"
+)
+
+var serviceIDs = []string{VPCServiceID, PVSServiceID, RCServiceID}
+
+// ServiceEndpoint holds the Service endpoint specific information.
+type ServiceEndpoint struct {
+	ServiceID     string
+	URL           string
+	ServiceRegion string
+}
+
+var (
+	errServiceEndpointFormat             = errors.New("must be formatted as ${ServiceID}=${URL}")
+	errServiceEndpointRegion             = errors.New("must be formatted as ${ServiceRegion}:${ServiceID1}=${URL1},${ServiceID2}=${URL2...}")
+	errServiceEndpointURL                = errors.New("must use a valid URL as a service-endpoint")
+	errServiceEndpointServiceID          = errors.New("must use a valid serviceID")
+	errServiceEndpointDuplicateServiceID = errors.New("same serviceID defined twice for service region")
+)
+
+// ParseServiceEndpointFlag parses the command line flag of service endpoint in the format ${ServiceRegion}:${ServiceID1}=${URL1},${ServiceID2}=${URL2...}
```

Comment:
As we have different region values we have for the vpc and powervs, e.g: lon for powervs but eu-gb for the vpc, in that case this may not work properly.

wondering if we can have following pattern to fix this:

```shell
${ServiceID1}=${ServiceRegion1}:${URL1},${ServiceRegion2}:${URL2};${ServiceID2}=${ServiceRegion1}:${URL1},${ServiceRegion2}:${URL2};...
```

Reply:
I think with the existing pattern following scenario should work fine (to-be-tested)

`lon:powervs=${URL1},rc=${URL2};eu-gb:vpc=${URL1}`